scripts/symbolicate-profile.ts

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env ts-node
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+/**
+ * This script helps to symbolicate a .cpuprofile file that was obtained from a specific (staging) app version (usually provided by a user using the app).
+ *
+ * @abstract
+ *
+ * 1. When creating a new deployment in our github actions, we upload the source map for android and iOS as artifacts.
+ * 2. The profiles created by the app on the user's device have the app version encoded in the filename.
+ * 3. This script takes in a .cpuprofile file, reads the app version from the filename, and downloads the corresponding source map from the artifacts using github's API.
+ * 4. It then uses the source map to symbolicate the .cpuprofile file using the `react-native-release-profiler` cli.
+ *
+ * @note For downloading an artefact a github token is required.
+ */
+import {Octokit} from '@octokit/core';
+import {execSync} from 'child_process';
+import fs from 'fs';
+import https from 'https';
+import path from 'path';
+import * as Logger from './utils/Logger';
+import parseCommandLineArguments from './utils/parseCommandLineArguments';
+
+const argsMap = parseCommandLineArguments();
+
+// #region Input validation
```

Comment:
What's with the `#region` / `#endregion` comments? Never seen these before.

imho if you want to make the sections of the script clearer, use comments in this style:

```ts
/* ============== INPUT VALIDATION ============== */
```

or even like this:

```ts
/* ==============================================
   ============== INPUT VALIDATION ============== 
   ============================================== */
```

Reply:
I guess it's a VSCode thing 🤷🏼